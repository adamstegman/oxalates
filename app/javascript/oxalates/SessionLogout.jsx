import React from 'react';
import PropTypes from 'prop-types';

import "./SessionLogout.scss";

export const SessionLogout = ({ onChangePassword }) => {
  const onLogOut = () => onChangePassword('');
  return (
    <button type="button"
            className="action logout"
            onClick={onLogOut}>
      Log out
    </button>
  );
};

SessionLogout.propTypes = {
  onChangePassword: PropTypes.func,
};
