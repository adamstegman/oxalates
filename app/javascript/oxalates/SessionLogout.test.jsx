import React from 'react';
import renderer from 'react-test-renderer';

import { SessionLogout } from './SessionLogout';

test('SessionLogout renders the logout button', () => {
  const component = renderer.create(
    <SessionLogout onLogout={() => {}} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
