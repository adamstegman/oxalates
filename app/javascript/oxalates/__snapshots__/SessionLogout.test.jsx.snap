// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SessionLogout renders the logout button 1`] = `
<button
  className="action logout"
  onClick={[Function]}
  type="button"
>
  Log out
</button>
`;
